packages/lesswrong/server/scripts/generativeModels/coverImages-2023Review.ts

```diff
@@ -0,0 +1,365 @@
+  const title = essay.title.replace(/[^a-zA-Z0-9]/g, '_').slice(0, 32);
+  const originId = encodeURIComponent(`${title}_${shortPrompt}_${Math.random()}`);
+  
+  const newUrl = await moveImageToCloudinary({oldUrl: url, originDocumentId: originId})
+  if (!newUrl) {
+    // eslint-disable-next-line no-console
+    console.error("Failed to upload image to cloudinary", {prompt, essay})
+    return
+  }
+  const reviewWinnerArt = await createMutator({
+    collection: ReviewWinnerArts,
+    context: createAdminContext(),
+    document: {
+      postId: essay.post._id, 
+      splashArtImagePrompt: prompt,
+      splashArtImageUrl: newUrl
+    }
+  })
+
+  // Add coordinates for the new art
+  if (reviewWinnerArt.data) {
+    await createMutator({
+      collection: SplashArtCoordinates,
+      context: createAdminContext(),
+      document: {
+        reviewWinnerArtId: reviewWinnerArt.data._id,
+        leftXPct: 0,
+        leftYPct: 0,
+        leftWidthPct: .33,
+        leftHeightPct: 1,
+        leftFlipped: false,
+        middleXPct: .33,
+        middleYPct: 0,
+        middleWidthPct: .33,
+        middleHeightPct: 1,
+        middleFlipped: false,
+        rightXPct: .66,
+        rightYPct: 0,
+        rightWidthPct: .33,
+        rightHeightPct: 1, 
+        rightFlipped: false,
+      }
+    });
+  }
+  
+  return reviewWinnerArt
+}
+
+const getEssaysWithoutEnoughArt = async (): Promise<Essay[]> => {
+  const targetAmountOfArt = 50
+
+  const postIds = await ReviewWinners.find({reviewYear: 2023}, { projection: { postId: 1, reviewYear: 1 }, sort: { reviewRanking: 1 } }).fetch();
+
+  const reviewArts = await ReviewWinnerArts.find({}).fetch();
+
+  const postIdsWithoutEnoughArt = postIds
+  .filter(p => reviewArts.filter(a => a.postId === p.postId).length < (targetAmountOfArt * .9))
+
+  const essays = await fetchFragment({
+    collectionName: "Posts",
+    fragmentName: "PostsPage",
+    selector: {_id: {$in: postIdsWithoutEnoughArt.map(p => p.postId)}},
+    currentUser: null,
+    skipFiltering: true,
+  });
+
+  return essays.map(e => {
+    const existingArtCount = reviewArts.filter(a => a.postId === e._id).length;
+    const neededArtCount = Math.max(0, targetAmountOfArt - existingArtCount);
+    return {post: e, title: e.title, content: e.contents?.html ?? "", neededArtCount, promptsGenerated: 10}
+  })
+}
+
+type Essay = {post: PostsPage, title: string, content: string, neededArtCount: number, promptsGenerated: number}
+
+const TextResponseFormat = zodResponseFormat(z.object({
+  metaphors: z.array(z.string()),
+}), "TextResponseFormat")
+
+const getPromptTextElements = async (openAiClient: OpenAI, essay: {title: string, content: string, promptsGenerated: number}, tryCount = 0): Promise<string[]> => {
+  const content = essay.content.length > 25_000 ? essay.content.slice(0, 12_500) + "\n[EXCERPTED FOR LENGTH]\n" + essay.content.slice(-12_500) : essay.content
```

Comment:
i'm not going to nitpick this file too much since it's a rarely-run script but this code is kinda hard to read